drivers/netdev_ieee802154/netdev_ieee802154.c

```diff
@@ -152,7 +164,14 @@ int netdev_ieee802154_get(netdev_ieee802154_t *dev, netopt_t opt, void *value,
 #endif
         case NETOPT_MAX_PDU_SIZE:
             assert(max_len >= sizeof(int16_t));
-            *((uint16_t *)value) = (IEEE802154_FRAME_LEN_MAX -
+
+            /* first determine the PHY mode */
+            res = dev->netdev.driver->get(&dev->netdev, NETOPT_IEEE802154_PHY, value, sizeof(uint8_t));
+            if (res < 0) {
+                *(uint8_t*)value = 0;
+            }
+
+            *((uint16_t *)value) = (_get_ieee802154_pdu(*(uint8_t*)value) -
```

Comment:
So `value` is just fetched to put it in this function. Why not make `netdev` the parameter for `_get_ieee802154_pdu()`? This would also make semantically more sense, as you typically request the PDU from the device, not the PHY mode ;-).

Reply:
Good idea!
But this is just a draft anyway. The more interesting question is: How (and where) do I inform GNRC about the change?